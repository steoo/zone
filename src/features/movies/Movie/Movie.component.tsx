import React, { FunctionComponent, useMemo } from 'react';
import { Background, Metadata, MovieContainer, VoteAverage } from '../Movies.styled';
import { IMAGE_BASE_URL } from '../../../app/constants';
import { Movie } from '../Movies.slice';

const MovieComponent: FunctionComponent<Movie> = ({ vote_average, poster_path, title, popularity }) => {
  const emoji = useMemo(() => {
    if (vote_average < 6) {
      return (
        <span role="img" aria-label="thinking">
          ðŸ¤”
        </span>
      );
    }

    if (vote_average < 7) {
      return (
        <span role="img" aria-label="smile">
          ðŸ™‚
        </span>
      );
    }

    if (vote_average <= 10) {
      return (
        <span role="img" aria-label="laugh">
          ðŸ˜ƒ
        </span>
      );
    }
  }, [vote_average]);

  return (
    <MovieContainer>
      <Background src={`${IMAGE_BASE_URL}${poster_path}`} />
      <VoteAverage>
        {vote_average} {emoji}
      </VoteAverage>

      <Metadata>
        <h1>{title}</h1>
        <p>Popularity: {popularity}</p>
      </Metadata>
    </MovieContainer>
  );
};

export default MovieComponent;
