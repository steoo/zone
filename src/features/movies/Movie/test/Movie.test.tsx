import React from 'react';
import { render } from '@testing-library/react';
import { movies } from '../../test/Movies.fixtures';

import Movie from '../Movie.component';

describe('Features', () => {
  describe('Movies', () => {
    describe('Component', () => {
      it('should render, without crashing', () => {
        const { baseElement } = render(<Movie {...movies[0]} />);

        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
