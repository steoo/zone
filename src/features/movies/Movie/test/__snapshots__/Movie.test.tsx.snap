// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Movies Component should render, without crashing 1`] = `
<body>
  <div>
    <div
      class="sc-gsTCUz erFNzH"
    >
      <div
        class="sc-hKgILt eciuch"
        src="https://image.tmdb.org/t/p/w200///8UlWHLMpgZm9bx6QYh0NFoq67TZ.jpg"
      />
      <div
        class="sc-dlfnbm eSOSpU"
      >
        7
         
        <span
          aria-label="laugh"
          role="img"
        >
          ðŸ˜ƒ
        </span>
      </div>
      <div
        class="sc-eCssSg fXmEff"
      >
        <h1>
          Wonder Woman 1984
        </h1>
        <p>
          Popularity: 
          2407.318
        </p>
      </div>
    </div>
  </div>
</body>
`;
